video/csputils.c

```diff
@@ -100,6 +100,22 @@ const struct m_opt_choice_alternatives mp_chroma_names[] = {
     {0}
 };
 
+void mp_colorspace_merge(struct mp_colorspace *orig, struct mp_colorspace *new)
+{
+    if (!orig->space)
+        orig->space = new->space;
+    if (!orig->levels)
+        orig->levels = new->levels;
+    if (!orig->primaries)
+        orig->primaries = new->primaries;
+    if (!orig->gamma)
+        orig->gamma = new->gamma;
+    if (!orig->nom_peak)
+        orig->nom_peak = new->nom_peak;
+    if (!orig->sig_peak)
+        orig->sig_peak = new->sig_peak;
+}
+
```

Comment:
That's the question. I think I even asked Matroska people to clarify whether container or codec information takes priority. Certainly it does happen that they conflict, such as with aspect ratio (which is basically a huge drama with every media player having different behavior).